globus_cli/services/transfer/async_delete.py

```diff
@@ -18,26 +18,24 @@
                      'asynchronous task.'))
 @common_options
 @submission_id_option
-@endpoint_id_option(help='ID of the Endpoint from which to delete file(s)')
-@click.option('--path', help='Path to the file/dir to delete')
 @click.option('--recursive', is_flag=True, help='Recursively delete dirs')
 @click.option('--ignore-missing', is_flag=True,
               help="Don't throw errors if the file or dir is absent")
 @click.option('--batch', is_flag=True,
               help=('Accept a batch of paths on stdin (i.e. run in '
-                    'batchmode). Uses --endpoint-id as passed on the '
-                    'commandline.'))
-def async_delete_command(batch, ignore_missing, recursive, path, endpoint_id,
-                         submission_id):
+                    'batchmode). Uses ENDPOINT_ID as passed on the '
+                    'commandline, and includes any commandline PATH given'))
+@click.argument('endpoint_plus_path', metavar=ENDPOINT_PLUS_OPTPATH.metavar,
+                type=ENDPOINT_PLUS_OPTPATH)
+def async_delete_command(batch, ignore_missing, recursive, submission_id,
+                         endpoint_plus_path):
     """
     Executor for `globus transfer async-delete`
     """
+    endpoint_id, path = endpoint_plus_path
     if path is None and (not batch):
```

Comment:
Also, as a sort of aside: what is the correct behavior when someone uses a path with batch mode? I thought we had talked about it prepending onto each line in the batch input (e.g. someone knows a base directory and wants to batch delete a bunch of junk inside of it), but the current state of affairs is to just add the path passed as an argument as another item to be deleted along with the items entered in the batch, which doesn't make any sense.

If users are actually NOT supposed to be able to call with a path in batch mode, maybe this should fail if both are set or both are unset (e.g. `if bool(path) is bool(batch): ... UsageError ...`).